daml/ContingentClaims/Claim.daml

```diff
@@ -114,33 +114,41 @@ isBearerExecutor = cata \case
 
 -- | Converts a `Serialized.Claim` read from ledger, to a more general `Claim`, for use with `Daml.Control.Recursion`.
 deserialize : Serialized.Claim t a -> Claim Observation t a
-deserialize = ana \case
-  Serialized.Zero -> ZeroF
-  Serialized.One a -> OneF a
-  Serialized.Give c -> GiveF c
-  Serialized.And c c' -> AndF c c'
-  Serialized.Or c c' -> OrF c c'
-  Serialized.Cond k c c' -> CondF k c c'
-  Serialized.Scale k c -> ScaleF k c
-  Serialized.When p c -> WhenF p c
---  Serialized.Anytime p c -> AnytimeF p c
---  Serialized.Until p c -> UntilF p c
+deserialize = ana deserialize' 
+
+-- | F-algebra for `deserialize`; it can be composed in unfolds.
+deserialize' : Serialized.Claim t a -> ClaimF Observation t a (Serialized.Claim t a)
+deserialize' Serialized.Zero = ZeroF
+deserialize' (Serialized.One a) = OneF a
+deserialize' (Serialized.Give c) = GiveF c
+deserialize' (Serialized.And [c]) = deserialize'  c
+deserialize' (Serialized.And (c :: cs)) = AndF c (Serialized.And cs)
+deserialize' (Serialized.And []) = error "deserialize: Malformed `And` (should have at least two elements)"
```

Comment:
you say "should have at least two elements" but you handle single elements above gracefully :)

Reply:
yes that's true ... what this means is the identity `serialize . deserialize == identity` holds, but `deserialize . serialize /= identity` in the case where you have a `And [c]` single element node (it will reduce to a `c`)